src/holodeck/agents.py

```diff
@@ -155,14 +153,16 @@ def teleport(self, location=None, rotation=None):
                 If ``None`` (default), keeps the current rotation.
 
         """
-        val = 0
-        if location is not None:
-            val += 1
+        if location is not None and rotation is None:
             np.copyto(self._teleport_buffer[0:3], location)
-        if rotation is not None:
+            self._teleport_type_buffer[0] = TeleportFlags.TELEPORT_LOCATION
+        if rotation is not None and location is None:
             np.copyto(self._teleport_buffer[3:6], rotation)
-            val += 2
-        self._teleport_type_buffer[0] = val
+            self._teleport_type_buffer[0] = TeleportFlags.TELEPORT_ROTATE
+        else:
+            np.copyto(self._teleport_buffer[0:3], location)
+            np.copyto(self._teleport_buffer[3:6], rotation)
+            self._teleport_type_buffer[0] = (TeleportFlags.TELEPORT_LOCATION | TeleportFlags.TELEPORT_ROTATE)
```

Comment:
Looking great so far. Just one more thing (🤞).

You can also boil the logic here down to something like this:

```python
if location is not None:
    np.copyto(self._teleport_buffer[0:3], location)
    self._teleport_type_buffer[0] |= TeleportFlags.TELEPORT_LOCATION
if rotation is not None:
    np.copyto(self._teleport_buffer[3:6], rotation)
    self._teleport_type_buffer[0] |= TeleportFlags.TELEPORT_ROTATE
```

I don't want to assume what you're familiar with, so here's a bit of CS 224. The `|` operator indicates a bitwise `OR`, which effectively goes bit-by-bit through both operands and picks the highest value. `_teleport_type_buffer[0]` ostensibly starts out with a binary representation of `00000000`, so if our flag's value is `00000010`, then the representation of `_teleport_type_buffer[0] |= flag` is `00000000 | 00000010 = 00000010`.